src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -198,22 +203,182 @@ std::optional<Values> Visitor::visitTypesafe(Parser::ValuesClauseContext* ctx) {
   return visitOptional(ctx->dataBlock());
 }
 
+// ____________________________________________________________________________________
+ParsedQuery::GraphPattern Visitor::visitTypesafe(
+    Parser::GroupGraphPatternContext* ctx) {
+  ParsedQuery::GraphPattern pattern;
+  if (ctx->subSelect()) {
+    pattern._children.emplace_back(visitTypesafe(ctx->subSelect()));
+    return pattern;
+  } else if (ctx->groupGraphPatternSub()) {
+    auto [subOps, filters] = visitTypesafe(ctx->groupGraphPatternSub());
+    pattern._children = std::move(subOps);
+    // TODO: moving
+    for (auto& filter : filters) {
+      if (filter._type == SparqlFilter::LANG_MATCHES) {
+        pattern.addLanguageFilter(filter._lhs, filter._rhs);
+      } else {
+        pattern._filters.push_back(std::move(filter));
+      }
+    }
+    return pattern;
+  } else {
+    AD_FAIL();  // Unreachable
+  }
+}
+
+// ____________________________________________________________________________________
+namespace {
+template <typename A, typename B>
+std::pair<vector<A>, vector<B>> splitUp(vector<std::variant<A, B>> elems) {
+  vector<A> as;
+  vector<B> bs;
+  auto aOp = [&as](A a) { as.push_back(std::move(a)); };
+  auto bOp = [&bs](B b) { bs.push_back(std::move(b)); };
+  for (auto& elem : elems) {
+    std::visit(ad_utility::OverloadCallOperator{aOp, bOp}, std::move(elem));
+  }
+  return {as, bs};
+}
+
+template <typename A, typename B>
+std::pair<vector<A>, vector<B>> splitUp(vector<std::pair<A, B>> elems) {
+  vector<A> as;
+  vector<B> bs;
+  for (auto [a, b] : elems) {
+    as.push_back(a);
+    bs.pusb_back(b);
+  }
+  return {as, bs};
+}
+}  // namespace
```

Comment:
Maybe these are part of the wrong rearranging of child patterns. I will look at them later, if they are still needed.

Reply:
They are no longer needed.